### ISSUE TITLE
Domain of rec:substance

### ISSUE THREAD

**Issue by ektrah** [RealEstateCore/REC-SHACL#15]
https://github.com/RealEstateCore/REC4/blob/86098485de50ef7c931b9bc656029822019b0dd4/Ontology/SHACL/RealEstateCore/rec.ttl#L1967-L1972

This seems to say that the domain of the `rec:substance` predicate are instances of the two classes, `rec:feeds` and `rec:isFedBy`. Is this interpretation correct? If yes, is it intended that `rec:feeds` and `rec:isFedBy` are classes?

**Comment by hammar** [comment]
Hi,

Nicely noticed! This is a case of unorthodox punning that we do to support emitting property graph constructs when translating SHACL -> DTDL. 

The intent is to symbolize that knowledge graph edges using the properties rec:isFedBy or rec:feeds can have the rec:substance property applied to them. So, the rdfs:domain statements above should really be understood to be not the properties themselves, but triples in which they are the predicate. 

Given that RDF does not allow for such constructs natively (at least not until RDF* is standardized), for now, one has to find workarounds. This is such a workaround. Suggestions on alternate representations are most welcome.

**Comment by ektrah** [comment]
Thanks for the explanation! It seems the common workarounds look like this:

![](https://www.ontotext.com/wp-content/uploads/2021/06/What-RDF-star-Improves.svg) ([source](https://www.ontotext.com/knowledgehub/fundamentals/what-is-rdf-star/))

Would one of these be option? (e.g., *N-ary Relations*)

**Comment by hammar** [comment]
Since REC4 is now merged into the [main REC repo](https://github.com/RealEstateCore/rec) we will be archiving this repository, and I'll close this issue. That doesn't imply that this discussion is dead -- would be happy to continue it in that other repo instead.

Unfortunately the proposed solutions would break alignment with Brick Schema, from which the feeds property is borrowed. My suggestion would be that we try and resolve this upstream, by contributions to / work with the Brick colleagues, who, it seems to me, would have ownership over this problem space.
